hack/test-go.sh

```diff
@@ -288,6 +288,9 @@ elif [[ -n "${coverage_output_dir}" ]]; then
     # we can clean up all of the coverage reports at once as they all exist in subdirectories of ${coverage_output_dir}/${OS_GO_PACKAGE}
     # and they are the only files found in those subdirectories
     rm -rf "${coverage_output_dir:?}/${OS_GO_PACKAGE}"
+
+elif [[ -n "${DEBUG-}" ]]; then
```

Comment:
Unless we don't want the 'set-but-null' behavior, let's stick to the more common defaulting substitution for uniformity

```
+elif [[ -n "${DEBUG:-}" ]]; then
                    ^
```